cpp/src/arrow/array-test.cc

```diff
@@ -58,6 +58,29 @@ TEST_F(TestArray, TestLength) {
   auto data = std::make_shared<PoolBuffer>(pool_);
   std::unique_ptr<Int32Array> arr(new Int32Array(100, data));
   ASSERT_EQ(arr->length(), 100);
+
+  // test slice functions 
+  auto mut_data = data;
```

Comment:
Indeed.
I should use the full name mutable_data instead of mut_data, I think.
Thanks for your advice, I will keep it in my mind.
:)